### PR TITLE
[QNN] InferType changes that missed CI.

### PR DESCRIPTION
**Pull request by anijain2305**
@vinx13 @ZihengJiang 

Added a quick fix to unblock CI. Will think about requantize default output_dtype to remove the dependence on InferType.

### CODE REVIEW

File: python/tvm/relay/qnn/op/qnn.py [hunk 1]
```diff
@@ -113,7 +113,10 @@ def concatenate(data,
 
     # Find the dtype of the input expr. This is required for the requantize op. Since, this is
     # concatenate op, the dtype of the input is same as dtype of the output.
-    data0 = relay.transform.infer_type(data[0])
+    mod = relay.Module.from_expr(data[0])
+    mod = relay.transform.InferType()(mod)
+    entry = mod["main"]
+    data0 = entry if isinstance(data[0], relay.Function) else entry.body
```
Comment (shoubhik) [inline python/tvm/relay/qnn/op/qnn.py]: Why not use `from ...frontend.common import infer_type`. It seems unnecessary duplication of code. If the namespace is a problem we should probably move the existing method to a better place?

Comment (shoubhik) [reply python/tvm/relay/qnn/op/qnn.py]: Also, it seems the method `from ...frontend.common import infer_type` has more checks.

Comment (ZihengJiang) [reply python/tvm/relay/qnn/op/qnn.py]: It seems strange to import pass like `infer_type` from `frontend`. Should we consider create shortcut for `infer_type`? @jroesch

Comment (anijain2305) [reply python/tvm/relay/qnn/op/qnn.py]: Shortcut seems like a good idea.

Maybe, we can get this in to unblock CI and then follow up with another PR for shortcut and related changes to remove code duplication.

**Pull request merged**